### PR TITLE
feat(DCS): add dcs template data sources

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dcs template data sources

1.  add dcs templates data source
2.  add dcs template detail data source

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.  add dcs templates data source
2.  add dcs template detail data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dcs/ TESTARGS='-run TestAccDatasourceTemplates_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccDatasourceTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceTemplates_basic
=== PAUSE TestAccDatasourceTemplates_basic
=== CONT  TestAccDatasourceTemplates_basic
--- PASS: TestAccDatasourceTemplates_basic (23.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       23.157s

make testacc TEST=./huaweicloud/services/acceptance/dcs/ TESTARGS='-run TestAccDatasourceTemplateDetail_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccDatasourceTemplateDetail_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceTemplateDetail_basic
=== PAUSE TestAccDatasourceTemplateDetail_basic
=== CONT  TestAccDatasourceTemplateDetail_basic
--- PASS: TestAccDatasourceTemplateDetail_basic (14.61s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       14.649s
```

### CODE REVIEW

File: docs/data-sources/dcs_template_detail.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `type` - (Required, String) Specifies the type of the template. Value options:
+  + **sys**: system template.
+  + **user**: custom template.
+
+* `template_id` - (Required, String) Specifies the ID of the template.
+
+* `params` - (Optional, List) Specifies the ID of the template.
```
Comment (deer-hang) [inline docs/data-sources/dcs_template_detail.md]:
```suggestion
* `params` - (Optional, List) Specifies the params of the template.
```

File: huaweicloud/services/dcs/data_source_huaweicloud_dcs_templates.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+		})
+	}
+	return rst
+}
+
+func filterGetDcsTemplatesResponseBodyTemplate(all []interface{}, d *schema.ResourceData) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	for _, v := range all {
+		if param, ok := d.GetOk("template_id"); ok &&
+			fmt.Sprint(param) != fmt.Sprint(utils.PathSearch("template_id", v, nil)) {
```
Comment (deer-hang) [inline huaweicloud/services/dcs/data_source_huaweicloud_dcs_templates.go:223]:
```suggestion
			fmt.Sprint(param) != fmt.Sprint(utils.PathSearch("template_id", v, "")) {
```

File: docs/data-sources/dcs_template_detail.md (hunk 1 above)
Comment (houpeng80) [reply docs/data-sources/dcs_template_detail.md]: done

File: huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_template_detail_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  template_id = data.huaweicloud_dcs_templates.test.templates[0].template_id
+  type        = "sys"
+}
+
+data "huaweicloud_dcs_template_detail" "params_filter" {
+  template_id = data.huaweicloud_dcs_templates.test.templates[0].template_id
+  type        = "sys"
+
+  params {
+    param_name  = "timeout"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_template_detail_test.go]:
```suggestion
    param_name = "timeout"
```

File: huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_templates_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+}
+output "name_filter_is_useful" {
+  value = length(data.huaweicloud_dcs_templates.name_filter.templates) > 0 && alltrue(
+    [for v in data.huaweicloud_dcs_templates.name_filter.templates[*].name : v == "Default-Redis-6.0-ha-enterprise-SSD"]
+  )  
+}
+
+data "huaweicloud_dcs_templates" "engine_filter" {
+  type = "sys"
+  engine = "Redis"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_templates_test.go]: format error, please check them all

```suggestion
  type   = "sys"
  engine = "Redis"
```

Comment (houpeng80) [reply huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_templates_test.go]: done

File: huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_template_detail_test.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dcs/data_source_huaweicloud_dcs_template_detail_test.go]: done

**Pull request merged**